pdp/handlers_upload.go

```diff
@@ -0,0 +1,285 @@
+package pdp
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+	logger "github.com/ipfs/go-log/v2"
+	"github.com/yugabyte/pgx/v5"
+
+	commcid "github.com/filecoin-project/go-fil-commcid"
+	commp "github.com/filecoin-project/go-fil-commp-hashhash"
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/curio/harmony/harmonydb"
+	"github.com/filecoin-project/curio/lib/dealdata"
+)
+
+var log = logger.Logger("pdp")
+
+// PieceSizeLimit in bytes
+var PieceSizeLimit = abi.PaddedPieceSize(256 << 20).Unpadded()
+
+func (p *PDPService) handlePiecePost(w http.ResponseWriter, r *http.Request) {
```

Comment:
This isn't a now problem, but I kind of forgot to mention that we really do all our uploads as traditional SHA256s of the actually blob of unpadded raw data, not PieceCIDs. We then calculate piece cids later for Filecoin. For demo purposes, we'll have our sidecar take in the data so we don't have to worry about this, and if we want to use the endpoint directly one day, we can discuss how to do this conversion.